components/script/dom/bluetooth.rs

```diff
@@ -119,7 +139,9 @@ fn convert_request_device_options(options: &RequestDeviceOptions,
     let mut optional_services = vec!();
     if let Some(ref opt_services) = options.optionalServices {
         for opt_service in opt_services {
-            optional_services.push(try!(BluetoothUUID::GetService(global, opt_service.clone())).to_string());
+            let uuid = try!(BluetoothUUID::GetService(global, opt_service.clone()));
+            return_if_blacklisted!(uuid, is_excluded);
+            optional_services.push(uuid.to_string());
```

Comment:
Yes if we rewrite the macro to return a result, then we can use a ``` try! ``` again to make the code shorter. But we discarded that change because that was confusing. See commit "Move blacklist check to the DOM side ":
``` 
optional_services.push(try!(
         return_if_blacklisted!(try!(BluetoothUUID::GetService(global, service.clone())).as_ref())));  ```
Now we need bind the variable because the change in the macro code.